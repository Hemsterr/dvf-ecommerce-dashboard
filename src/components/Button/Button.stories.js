// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

// Components
import Button from './index'

storiesOf('Button', module)
  .add('Default', () => (
    <Button label="default" handleOnClick={action('clicked')} />
  ))
  .add('Disabled', () => (
    <Button label="Disabled" handleOnClick={action('clicked')} disabled />
  ))
  .add('Danger', () => (
    <Button
      className="btn btn__danger"
      label="Danger"
      handleOnClick={action('clicked')}
    />
  ))
  .add('Success', () => (
    <Button
      className="btn btn__success"
      label="Success"
      handleOnClick={action('clicked')}
    />
  ))
  .add('Primary', () => (
    <Button
      className="btn btn__primary"
      label="Primary"
      handleOnClick={action('clicked')}
    />
  ))
  .add('Size small', () => (
    <Button
      className="btn btn__basic btn__small"
      label="small"
      handleOnClick={action('clicked')}
    />
  ))
  .add('Size medium', () => (
    <Button
      className="btn btn__basic btn__medium"
      label="medium"
      handleOnClick={action('clicked')}
    />
  ))
  .add('Size lager', () => (
    <Button
      className="btn btn__basic btn__lager"
      label="lager"
      handleOnClick={action('clicked')}
    />
  ))
