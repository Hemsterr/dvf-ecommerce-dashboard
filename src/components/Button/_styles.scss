.btn {
  color: $alabaster;
  font-family: $font-base-bold;
  text-align: center;
  border: none;
  cursor: pointer;
  &:disabled {
    opacity: 0.7;
    cursor: unset;
  }

  &:focus {
    outline: none;
  }

  &__small {
    @include text-size(pxToRem(12px), pxToRem(2px), pxToRem(30px));
    height: pxToRem(35px);
  }

  &__medium {
    @include text-size(pxToRem(15px), pxToRem(4.5px), pxToRem(20px));
    height: pxToRem(40px);
    padding: 0 pxToRem(50px);
    font-family: $font-base-semi-bold;
    text-transform: uppercase;
  }

  &__lager {
    @include text-size(pxToRem(15px), pxToRem(4px), pxToRem(20px));
    height: pxToRem(50px);
  }

  &__primary {
    background-color: $shuttle-gray;
  }

  &__danger {
    background-color: $silver-rust;
  }

  &__success {
    background-color: $big-stone;
  }
}
