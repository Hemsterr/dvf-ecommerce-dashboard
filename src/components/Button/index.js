// @flow
import React from 'react'

type Props = {
  label: string,
  handleOnClick: Function,
  className: string,
  disabled: boolean,
  size: string,
}

class Button extends React.PureComponent<Props> {
  static defaultProps = {
    label: '',
    className: '',
    disabled: false,
    handleOnClick: () => {},
    size: 'medium',
  }

  render() {
    const {
      label, className, handleOnClick, disabled, size
    } = this.props

    return (
      <button
        className={`${className} btn__${size}`}
        onClick={handleOnClick}
        disabled={disabled}
        type="button"
      >
        {label}
      </button>
    )
  }
}

export default Button
