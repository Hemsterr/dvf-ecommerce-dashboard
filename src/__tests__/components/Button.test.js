// Libs
import React from 'react'
import { render, fireEvent } from '@testing-library/react'

// Component
import Button from '../../components/Button'

const props = {
  label: 'Search',
  handleOnClick: jest.fn(),
  className: 'btn btn__basic',
}

describe('Button component', () => {
  test('Render the button without props', () => {
    const { container } = render(<Button />)
    expect(container).toMatchSnapshot()
    expect(container.querySelectorAll('.btn__medium').length).toEqual(1)
    expect(Button.defaultProps.handleOnClick()).toBeUndefined()
  })

  test('Render the button with props', () => {
    const { container, rerender } = render(<Button {...props} />)

    // Checking event click button
    fireEvent.click(container)
    rerender(<Button {...props} />)
    expect(props.handleOnClick).toBeTruthy()
    expect(container.textContent).toEqual(props.label)
  })
})
