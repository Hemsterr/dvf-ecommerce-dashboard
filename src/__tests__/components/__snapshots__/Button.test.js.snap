// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Render the button without props 1`] = `
<div>
  <button
    class=" btn__medium"
    type="button"
  />
</div>
`;
